text/0032-fog-compatible-gift-codes.md

```diff
@@ -0,0 +1,255 @@
+When this TxOut is spent, the app can infer that the gift code was redeemed, or canceled. The app may cancel a gift code by
+paying it back to the primary subaddress.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In our opinion, there is no reason why not to adopt at least the `global_index` change, since this is superior in every way to
+sending the `tx_out_public_key`.
+
+Using the `TxOutGiftCode` instead of the transfer-payload in mobile apps is superior because it enables linked devices to track
+and potentially reclaim the gift code more finely, without relying on off-chain synchronized storage to track the in-flight transfer payloads.
+
+One drawback is that this redesign did not achieve the goal of having only one on-chain event per gift code.
+We had hoped that by eliminating the need for a temporary account, and just transfering a `TxOut` directly, we
+would make it so that only the recipient needs to have an on-chain event. It turns out that an on-chain event
+is still required in practice and probably unavoidable, at least with this approach.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There are a few minor technical variations we considered like, not revealing the shared secret, and revealing
+the amount and blinding factor directly instead, but this is more bytes on the wire, and it may be useful for
+the recipient to be able to read the memo anyways.
+
+Conceiveably, we could avoid the sender-side on-chain event, by using an [MCIP #31](https://github.com/mobilecoinfoundation/mcips/pull/0032)
+"Signed Contingent Input". Then if you have a `TxOut` for 100 MOB, and you want to create a gift-code for 10 MOB, you could make the signed
+contingent input and require that 90 MOB comes back to you as change, and give this to the counterparty, and then that would be the gift code.
+However, this has a lot of drawbacks, one being that, it doesn't help linked devices synchronize this, and another being that, these inputs
+are much bigger than the `TransferPayload` and the `TxOutGiftCode`, because they have to contain an entire Ring MLSAG.
+
+# Prior art
+[prior-art]: #prior-art
+
+None that we are aware of.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+There should probably be RTH memos that can be used to remember who you sent a gift code to,
+and help you remember that you canceled a gift code.
+
+We should make such memos in this MCIP or a later MCIP.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+It seems that we are often in a position of reserving subaddresses for new purposes discovered in mobile applications.
+
+This conflicts with how subaddresses are used in applications like `mobilecoind` and `full-service`, which are often assigning
+a new subaddress per contact.
+
+Since we want for users to be able to export their account from mobile apps and load them in desktop apps and have things work cleanly,
```

Comment:
Not sure how much we want to try and keep _all_ MCIPs up to date, especially when multiple associated ones end up in flight at the same time.
This could reference MCIP [36](https://github.com/mobilecoinfoundation/mcips/blob/main/text/0036-reserved-subaddresses.md) or perhaps this future possibilities section should go away.

I'm also fine with leaving this as is if that is desired.

Reply:
FWIW I think we should try to keep all the MCIPs up to date, so that readers who jump into our ecosystem can understand what's going on

Reply:
Given that I would vote for removing the future work section from this MCIP since it has now been accepted as [MCIP 36](https://github.com/mobilecoinfoundation/mcips/blob/main/text/0036-reserved-subaddresses.md)

Reply:
thank you